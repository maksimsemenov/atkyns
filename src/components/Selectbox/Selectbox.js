import React, { PropTypes } from 'react'
import l from 'utils/local'
import './Selectbox.less'

const Selectbox = ({ options, active = '', onChange }) => {
  const activeOptions = options.filter(option => !option.disabled)
  active = activeOptions.includes(active) ? active : activeOptions[0]
  return (
    <select
      className='selectbox'
      value={active}
      onChange={(e) => onChange(e.target.options.item(e.target.selectedIndex).value)}
    >
      {options.map((option, index) =>
        <option
          key={index}
          value={option.value}
          disabled={option.disabled}
        >{l(`%${option.value}`)}</option>
      )}
    </select>
  )
}

Selectbox.propTypes = {
  options: PropTypes.arrayOf(PropTypes.shape({
    value: PropTypes.string.isRequired,
    disabled: PropTypes.bool
  })).isRequired,
  active: PropTypes.string,
  onChange: PropTypes.func.isRequired
}

export default Selectbox
