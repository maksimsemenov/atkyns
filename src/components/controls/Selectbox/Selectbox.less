@import '~constants/styles.less';

.selectbox__control {
  width: 350px;
  max-width: 100%;
}
.selectbox {
  position: relative;
  z-index: 1;
  appearance: none;
  width: 100%;
  box-sizing: border-box;
  padding: 10px 10px 12px;

  border-radius: @radius-control;
  background-color: #ffffff;
  background-image: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" width="16px" height="10px" viewBox="0 0 16 10" fill="none"><polyline stroke="@{font-color-light}" stroke-width="1" points="1,1 8,9 15,1"/></svg>');
  background-repeat: no-repeat;
  background-position: ~"calc(100% - 12px)" center;
  color: @font-color-light;
  border: 1px solid @color-gray;
  font-size: @font-size-form;
  font-weight: @font-weight-basic;

  transition: border-color @transition-basic;

  &:hover {
    border-color: @color-dark-gray;
    transition: border-color @transition-active;
  }
  &:active,
  &:focus {
    border-color: @color-accent;
    outline: none;
    box-shadow: 0 0 3px 0 fade(@color-accent, 30%);
    transition: border-color @transition-active;
  }
  &.has-value {
    color: @font-color-basic;
  }
  &.has-error {
    border-color: @color-red;
    box-shadow: 0 0 3px 0 fade(@color-red, 30%);
  }

  .stage__fieldColumn &,
  .stage__fieldRow & {
    border-radius: 0;

    &:hover {
      z-index: 2;
    }
    &:active,
    &:focus {
      z-index: 3;
    }
  }
  .stage__fieldRow.is-first &.is-first {
    border-top-left-radius: @radius-control;
  }
  .stage__fieldRow.is-last &.is-first {
    border-bottom-left-radius: @radius-control;
  }
  .stage__fieldRow.is-first &.is-last {
    border-top-right-radius: @radius-control;
  }
  .stage__fieldRow.is-last &.is-last {
    border-bottom-right-radius: @radius-control;
  }
}

.selectbox__error {
  padding: 2px;
  line-height: 1.4;
  color: @color-red;
  font-size: @font-size-small;
}
.selectbox__note {
  padding: 2px;
  line-height: 1.4;
  color: @font-color-light;
  font-size: @font-size-small;
}
