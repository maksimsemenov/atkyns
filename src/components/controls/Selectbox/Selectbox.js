import React, { PropTypes } from 'react'
import classNames from 'classnames'
import l from 'utils/local'
import './Selectbox.less'

const Selectbox = ({ options, value = '', placeholder, onChange, error, note, className, ...rest }) => {
  const selectClasses = classNames({
    'selectbox': true,
    'has-value': value,
    'has-error': error,
    [className]: className
  })
  return (
    <div className='selectbox__control'>
      <select
        className={selectClasses}
        value={value}
        onChange={(e) => onChange(e.target.value)}
        {...rest}
      >
        {placeholder && <option value=''>{placeholder}</option>}
        {options.map((option, index) =>
          <option
            key={index}
            value={option.value}
            disabled={option.disabled}
          >{option.name}</option>
        )}
      </select>
      {error ? <div className='selectbox__error'>{l(error)}</div> : null}
      {!error && note && <div className='selectbox__note'>{l(note)}</div>}
    </div>
  )
}

Selectbox.propTypes = {
  options: PropTypes.arrayOf(PropTypes.shape({
    value: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    disabled: PropTypes.bool
  })).isRequired,
  value: PropTypes.string,
  placeholder: PropTypes.string,
  error: PropTypes.string,
  note: PropTypes.string,
  className: PropTypes.string,
  onChange: PropTypes.func.isRequired
}

export default Selectbox
