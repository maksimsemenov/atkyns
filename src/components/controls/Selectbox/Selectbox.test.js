import React from 'react'
import renderer from 'react-test-renderer'
import Selectbox from './Selectbox'

describe('Selectbox component', () => {
  it('Snapshot test', () => {
    const options = [
      { name: 'Option 1', value: 'option1' },
      { name: 'Option 2', value: 'option2' },
      { name: 'Option 3', value: 'option3' }
    ]
    const value = 'option1'
    const onChange = () => {}
    const params = { options, value, onChange}
    const tree = renderer.create(<Selectbox {...params} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
