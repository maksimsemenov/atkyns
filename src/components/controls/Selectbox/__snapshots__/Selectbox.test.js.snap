exports[`Selectbox component Snapshot test 1`] = `
<div
  className="selectbox__control"
  style={undefined}>
  <select
    className="selectbox has-value"
    onChange={[Function onChange]}
    value="option1">
    <option
      disabled={undefined}
      value="option1">
      Option 1
    </option>
    <option
      disabled={undefined}
      value="option2">
      Option 2
    </option>
    <option
      disabled={undefined}
      value="option3">
      Option 3
    </option>
  </select>
</div>
`;
